### PR TITLE
feat: Add AllowPrivateNetwork to support Access-Control-Allow-Private-Network

### PR DESCRIPTION
**Pull request by lvyanru8200**
Fixes: https://github.com/projectcontour/contour/issues/4553
Signed-off-by: tigerK <yanru.lv@daocloud.io>

### CODE REVIEW

**Comment by lvyanru8200** [comment]
cc @skriss @stevesloka

**Comment by skriss** [comment]
Thanks for the PR @lvyanru8200! Please add a changelog file per the instructions in https://github.com/projectcontour/contour/actions/runs/4068330638/jobs/7006764495. You'll also need to run `make generate` and commit the changed files.

File: internal/featuretests/v3/corspolicy_test.go [hunk 1]
```diff
@@ -331,9 +337,52 @@ func TestCorsPolicy(t *testing.T) {
...
+					&envoy_route_v3.Route{
+						Match:  routePrefix("/"),
+						Action: routecluster("default/svc1/80/da39a3ee5e"),
+					}),
+			),
+		),
+		TypeUrl: routeType,
+	})
+
+	//
```
Comment (skriss) [inline internal/featuretests/v3/corspolicy_test.go:354]: Fill in comment here 😁

**Comment by lvyanru8200** [comment]
cc @skriss

**Comment by skriss** [comment]
Looks like a few more failing unit tests to update.

**Comment by lvyanru8200** [comment]
cc @skriss

**Review by skriss — CHANGES_REQUESTED** [review]
Code LGTM. The last thing is to update the docs (https://github.com/projectcontour/contour/blame/main/site/content/docs/main/config/cors.md)

I think mentioning this in https://github.com/projectcontour/contour/blame/main/site/content/docs/main/config/cors.md#L7, plus adding it to one of the YAML examples with a sentence or comment explaining it, should be sufficient.

**Comment by lvyanru8200** [comment]
cc @skriss

**Review by skriss — APPROVED** [review]
LGTM, thanks @lvyanru8200.

**Pull request merged**